proxy/proxy.go

```diff
@@ -84,25 +84,27 @@ func NewProxy(
 
 	roundTripperFactory := &round_tripper.FactoryImpl{
 		BackendTemplate: &http.Transport{
-			DialContext:           dialer.DialContext,
-			DisableKeepAlives:     cfg.DisableKeepAlives,
-			MaxIdleConns:          cfg.MaxIdleConns,
-			IdleConnTimeout:       90 * time.Second, // setting the value to golang default transport
-			MaxIdleConnsPerHost:   cfg.MaxIdleConnsPerHost,
-			DisableCompression:    true,
-			TLSClientConfig:       backendTLSConfig,
-			TLSHandshakeTimeout:   cfg.TLSHandshakeTimeout,
-			ExpectContinueTimeout: 1 * time.Second,
+			DialContext:            dialer.DialContext,
+			DisableKeepAlives:      cfg.DisableKeepAlives,
+			MaxIdleConns:           cfg.MaxIdleConns,
+			IdleConnTimeout:        90 * time.Second, // setting the value to golang default transport
+			MaxIdleConnsPerHost:    cfg.MaxIdleConnsPerHost,
+			DisableCompression:     true,
+			TLSClientConfig:        backendTLSConfig,
+			TLSHandshakeTimeout:    cfg.TLSHandshakeTimeout,
+			ExpectContinueTimeout:  1 * time.Second,
+			MaxResponseHeaderBytes: int64(cfg.MaxResponseHeaderBytes),
 		},
 		RouteServiceTemplate: &http.Transport{
-			DialContext:           dialer.DialContext,
-			DisableKeepAlives:     cfg.DisableKeepAlives,
-			MaxIdleConns:          cfg.MaxIdleConns,
-			IdleConnTimeout:       90 * time.Second, // setting the value to golang default transport
-			MaxIdleConnsPerHost:   cfg.MaxIdleConnsPerHost,
-			DisableCompression:    true,
-			TLSClientConfig:       routeServiceTLSConfig,
-			ExpectContinueTimeout: 1 * time.Second,
+			DialContext:            dialer.DialContext,
+			DisableKeepAlives:      cfg.DisableKeepAlives,
+			MaxIdleConns:           cfg.MaxIdleConns,
+			IdleConnTimeout:        90 * time.Second, // setting the value to golang default transport
+			MaxIdleConnsPerHost:    cfg.MaxIdleConnsPerHost,
+			DisableCompression:     true,
+			TLSClientConfig:        routeServiceTLSConfig,
+			ExpectContinueTimeout:  1 * time.Second,
+			MaxResponseHeaderBytes: int64(cfg.MaxResponseHeaderBytes),
```

Comment:
just because the ports were recently changed to uint, should we have uint here as well?

Reply:
Well the transport is from the go standard library, we can't change the type there. I don't see much value in using a uint over a regular int, you won't be able to set a higher value because the stdlib only accepts signed ints and we just consider negative values as no limit (which means some default is put in place).

Reply:
now I see that this is a golang builtin. Disregard my comment.